File: server/php/src/Infrastructure/Mappers/PaymentDetails/SurchargeRateMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\SurchargeRate as SurchargeRateDto;
use OnlinePayments\Sdk\Domain\SurchargeRate as SurchargeRateSdk;

final class SurchargeRateMapper
{
    public static function mapFromSdkResponse(?SurchargeRateSdk $response): ?SurchargeRateDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new SurchargeRateDto();
        $dto->specificRate = $response?->specificRate ?? null;
        $dto->adValoremRate = $response?->adValoremRate ?? null;
        $dto->surchargeProductTypeVersion = $response?->surchargeProductTypeVersion ?? null;
        $dto->surchargeProductTypeId = $response?->surchargeProductTypeId ?? null;

        return $dto;
    }
}

File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/SurchargeRateMapper.cs
using SurchargeRateSdk = OnlinePayments.Sdk.Domain.SurchargeRate;
using SurchargeRateDto = Business.Domain.Payments.PaymentDetails.SurchargeRate;

namespace Infrastructure.Mappers.PaymentDetails;

public static class SurchargeRateMapper
{
    public static SurchargeRateDto Map(SurchargeRateSdk? response)
    {
        return new()
        {
            SpecificRate = response?.SpecificRate,
            AdValoremRate = response?.AdValoremRate,
            SurchargeProductTypeVersion = response?.SurchargeProductTypeVersion,
            SurchargeProductTypeId = response?.SurchargeProductTypeId,
        };
    }
}